src/test/java/seedu/findvisor/model/person/TagsContainsKeywordsPredicateTest.java

```diff
@@ -0,0 +1,70 @@
+package seedu.findvisor.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.findvisor.model.tag.TagsContainsKeywordsPredicate;
+import seedu.findvisor.testutil.PersonBuilder;
+
+public class TagsContainsKeywordsPredicateTest {
+
+    @Test
+    public void equals() {
+        List<String> firstPredicateKeywords = Arrays.asList(new String[]{"tag1"});
+        List<String> secondPredicateKeywords = Arrays.asList(new String[]{"tag2", "tag3"});
+
+        TagsContainsKeywordsPredicate firstPredicate = new TagsContainsKeywordsPredicate(firstPredicateKeywords);
+        TagsContainsKeywordsPredicate secondPredicate = new TagsContainsKeywordsPredicate(secondPredicateKeywords);
+
+        // same object -> returns true
+        assertTrue(firstPredicate.equals(firstPredicate));
+
+        // same values -> returns true
+        TagsContainsKeywordsPredicate firstPredicateCopy = new TagsContainsKeywordsPredicate(firstPredicateKeywords);
+        assertTrue(firstPredicate.equals(firstPredicateCopy));
+
+        // different types -> returns false
+        assertFalse(firstPredicate.equals(1));
+
+        // null -> returns false
+        assertFalse(firstPredicate.equals(null));
+
+        // different email -> returns false
```

Comment:
Should this be tags instead?
```suggestion
        // different tags-> returns false
```